File: tasks/json/medium/processJSON.js
const fs = require("fs");

const data = JSON.parse(fs.readFileSync("data.json", "utf-8"));

const updatedData = data
  .filter(item => item.status === "active")
  .map(item => ({
    ...item,
    price: item.price * 1.1
  }));

fs.writeFileSync("updated_data.json",
  JSON.stringify(updatedData, null, 2)
);
console.log("Updated_data.json create successfully");

File: tasks/json/medium/processJSON.py
import json


def main():
    with open("data.json", "r") as f:
        data = json.loads(f.read())

    products = [product for product in data if product["status"] != "inactive"]

    for product in products:
        product["price"] = int(product["price"] * 1.1)

    with open("updated_data.json", "w") as f:
        f.write(json.dumps(products, indent=4))


if __name__ == "__main__":
    main()
